Velaptor/Content/Caching/SoundCache.cs

```diff
@@ -0,0 +1,184 @@
+        private readonly ConcurrentDictionary<string, ISound> sounds = new ();
+        private readonly ISoundFactory soundFactory;
+        private readonly IFile file;
+        private readonly IPath path;
+        private readonly IDisposable shutDownUnsubscriber;
+        private readonly IReactable<DisposeSoundData> disposeSoundsReactable;
+        private bool isDisposed;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SoundCache"/> class.
+        /// </summary>
+        /// <param name="soundFactory">Creates <see cref="ISound"/> objects.</param>
+        /// <param name="file">Performs file related operations.</param>
+        /// <param name="path">Provides path related services.</param>
+        /// <param name="disposeSoundsReactable">Sends push notifications to dispose of sounds.</param>
+        /// <param name="shutDownReactable">Sends a push notifications that the application is shutting down.</param>
+        public SoundCache(
+            ISoundFactory soundFactory,
+            IFile file,
+            IPath path,
+            IReactable<DisposeSoundData> disposeSoundsReactable,
+            IReactable<ShutDownData> shutDownReactable)
+        {
+            this.soundFactory = soundFactory ?? throw new ArgumentNullException(nameof(soundFactory), NullParamExceptionMsg);
+            this.file = file ?? throw new ArgumentNullException(nameof(file), NullParamExceptionMsg);
+            this.path = path ?? throw new ArgumentNullException(nameof(path), NullParamExceptionMsg);
+
+            if (shutDownReactable is null)
+            {
+                throw new ArgumentNullException(nameof(shutDownReactable), NullParamExceptionMsg);
+            }
+
+            this.shutDownUnsubscriber = shutDownReactable.Subscribe(new Reactor<ShutDownData>(_ => ShutDown()));
+
+            this.disposeSoundsReactable =
+                disposeSoundsReactable ??
+                throw new ArgumentNullException(nameof(disposeSoundsReactable), NullParamExceptionMsg);
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the <see cref="SoundCache"/> class.
+        /// </summary>
+        [ExcludeFromCodeCoverage]
+        ~SoundCache()
+        {
+            if (UnitTestDetector.IsRunningFromUnitTest)
+            {
+                return;
+            }
+
+            ShutDown();
+        }
+
+        /// <inheritdoc/>
+        public int TotalCachedItems => this.sounds.Count;
+
+        /// <inheritdoc/>
+        public ReadOnlyCollection<string> CacheKeys => new (this.sounds.Keys.ToArray());
+
+        /// <summary>
+        /// Gets a sound using the given <paramref name="soundFilePath"/>.
+        /// </summary>
+        /// <param name="soundFilePath">
+        ///     The full file path to a <c>Sound</c> file.
+        /// </param>
+        /// <returns>A sound loaded from either a <c>'.ogg'</c> or <c>'.mp3'</c> file.</returns>
```

Comment:
Refactor `<returns>` code doc

Reason: Grammar issue

```suggestion
        /// <returns>A sound loaded from either an <c>'.ogg'</c> or <c>'.mp3'</c> file.</returns>
```